sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Expression.scala

```diff
@@ -89,6 +89,12 @@ abstract class BinaryExpression extends Expression with trees.BinaryNode[Express
   override def toString: String = s"($left $symbol $right)"
 }
 
+abstract class BinaryFunctionExpression extends Expression with trees.BinaryNode[Expression] {
```

Comment:
can you document what's the difference between BinaryFunctionExpression and BinaryExpression? Before the addition of this function everything was pretty self evident.